elks/arch/i86/drivers/char/console-bios.c

```diff
@@ -111,14 +118,14 @@ static void scroll(register Console * C, int n, int x, int y, int xx, int yy)
     int a;
 
     a = C->attr;
-    if (C != Visible) {
+    if (C != Visible[C->op]) {
```

Comment:
I think here we have a situation similar to glock but in reverse: we don't need a global Visible, it seems (nor a "op" member specially to index it). What we likely need is a "char visible" member that tells whether the given console is active (visible) or not. This member can then be used to determine whether to display the incoming data etc, or just store it, depending on whether the specified console is visible or not. Your driver would have to be responsible for setting the visible member correctly for both displays, or only a single display in the non DUAL case. The driver probably should have been written this way in the first place, except it was using globals for most everything - so another was added.

See what I mean?

Reply:
In other words, we don't need an 'op' member to index a global outside struct Console *Visible[] array, to get another Console *; instead, we just keep a new Console struct member that allows the routine to perform the output or not, as there isn't a need to actually reference a different Console * than the one passed.

Reply:
With the latest changes, I think I can now see why you're still using a Visible[] array: there are two displays! So that's fine, changes look good.

Reply:
I see what you mean, it's fairly useless in most places, and was only kept like this because `console-bios.c`, does some things with it that I didn't fully work out yet as I was working on other things.
I'll see if I can factor it out tomorrow.

Reply:
Is the `console-bios.c` required on any platform, by the way? Would removing it entirely be feasible, if it's been mostly replaced by `console-direct.c`?

Reply:
console-bios.c is definitely required - its used on all those PC-like systems that won't run console-direct.c or don't have a 6845. We also have some skillful players in the past, that have embedded ELKS into ROM, but want a full-featured console, and implemented INT 10 functions so that console-bios could run.

We try to keep ELKS dependencies to a bare minimum, as we actually have a wide variety of use cases being one of the few actively worked on, stable 8086 operating systems around.